crates/uniswapx-rs/src/order.rs

```diff
@@ -29,30 +32,79 @@ sol! {
     }
 
     #[derive(Debug)]
-    struct ExclusiveDutchOrder {
-        OrderInfo info;
+    struct CosignerData {
         uint256 decayStartTime;
         uint256 decayEndTime;
         address exclusiveFiller;
         uint256 exclusivityOverrideBps;
-        DutchInput input;
-        DutchOutput[] outputs;
+        uint256 inputAmount;
+        uint256[] outputAmounts;
+    }
+
+    #[derive(Debug)]
+    struct V2DutchOrder {
+        OrderInfo info;
+        address cosigner;
+        DutchInput baseInput;
+        DutchOutput[] baseOutputs;
+        CosignerData cosignerData;
+        bytes cosignature;
+    }
+    
+    #[derive(Debug)]
+    struct PriorityInput {
+        address token;
+        uint256 amount;
+        uint256 mpsPerPriorityFeeWei;
+    }
+
+    #[derive(Debug)]
+    struct PriorityOutput {
+        address token;
+        uint256 amount;
+        uint256 mpsPerPriorityFeeWei;
+        address recipient;
+    }
+
+    #[derive(Debug)]
+    struct PriorityCosignerData {
+        uint256 auctionTargetBlock;
+    }
+
+    #[derive(Debug)]
+    struct PriorityOrder {
+        OrderInfo info;
+        address cosigner;
+        uint256 auctionStartBlock;
+        uint256 baselinePriorityFeeWei;
+        PriorityInput input;
+        PriorityOutput[] outputs;
+        PriorityCosignerData cosignerData;
+        bytes cosignature;
     }
 }
 
-pub fn decode_order(encoded_order: &str) -> Result<ExclusiveDutchOrder> {
-    let encoded_order = if encoded_order.starts_with("0x") {
-        &encoded_order[2..]
-    } else {
-        encoded_order
-    };
-    let order_hex = hex::decode(encoded_order)?;
 
-    Ok(ExclusiveDutchOrder::decode(&order_hex, false)?)
+#[derive(Debug, Clone)]
+pub enum Order {
+    V2DutchOrder(V2DutchOrder),
+    PriorityOrder(PriorityOrder),
 }
 
-pub fn encode_order(order: &ExclusiveDutchOrder) -> Vec<u8> {
-    ExclusiveDutchOrder::encode(order)
+impl Order {
+    pub fn resolve(&self, block_number: u64, timestamp: u64, priority_fee: Uint<256, 4>) -> OrderResolution {
+        match self {
+            Order::V2DutchOrder(order) => order.resolve(timestamp),
+            Order::PriorityOrder(order) => order.resolve(block_number, timestamp, priority_fee),
```

Comment:
given these take such different parameters, i think I'd lean to not including it at this level but making the caller match itself. ie if only dealing with V2DutchOrders it doesn't seem worth it to have to pass block & priority fee to resolve